File: Domain/Job.cs
using System;
using System.Collections.Generic;

namespace Domain
{
    public class Job
    {
        public Guid Id { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public string  Category { get; set; }
        public string City { get; set; }
        public string WorkingHours { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime ExpiresAt { get; set; }

        public virtual ICollection<UserJob> UserJobs { get; set; }
    }
}

File: client-app/src/app/models/job.ts
export interface IJob {
	id: string;
	title: string;
	description: string;
	category: string;
	workingHours: string;
	city: string;
	createdAt: Date;
  expiresAt: Date;
	isOwner: boolean;
	attendees: IAttendee[];
}

export interface IJobFormValues extends Partial<IJob> {
  time?: Date;
}

export class JobFormValues implements IJobFormValues {
	id?: string = undefined;
	title: string = '';
	category: string = '';
	description: string = '';
	createdAt?: Date = undefined;
  expiresAt?: Date = undefined;
	time?: Date = undefined;
	city: string = '';
	workingHours?: string = undefined;

	constructor(init?: IJobFormValues) {
    if(init && init.expiresAt){
      init.time = init.expiresAt;
    }
		Object.assign(this, init);
	}
}

export interface IAttendee {
	username: string;
	displayName: string;
	image: string;
	isOwner: boolean;
	phoneNumber: string;
	city: string;
}